internal/epoch/epoch_manager.go

```diff
@@ -950,6 +1023,101 @@ func (e *Manager) getCompleteIndexSetForCommittedState(ctx context.Context, cs C
 	return result, nil
 }
 
+type intRange struct {
+	lo, hi int
+}
+
+func (r intRange) length() int {
+	return r.hi - r.lo
+}
+
+func (r intRange) isEmpty() bool {
+	return r.length() <= 0
+}
+
+var errNonContiguousRange = errors.New("non-contiguous range")
+
+// Returns a continuous close-open epoch range for the keys, that is [lo, hi).
+// A range of the form [v,v) means the range is empty.
+// When the range is not continuous an error is returned.
+func getKeyRange[E any](m map[int]E) (intRange, error) {
```

Comment:
(organization) Totally optional, but this implementation seems generic enough to be decomposed into its own helper package or file, along with its unit test, perhaps?

Reply:
Yeah, we can do that in a separate PR.

Also, it'd be good to discuss a broader philosophical question here, since it is not clear to me what's the _threshold or breaking point_ between:
- Reuse (_"don't repeat yourself"_, etc) and
- _"A little copying is better than a little dependency."_

Anyway, let's discuss that separate from this PR.

Reply:
For now, it's been moved it to the `_utils` file in the same package.